flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/TableSchemaUtils.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.utils;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.util.Preconditions;
+
+/**
+ * Utilities to filter generated column from {@link TableSchema}.
+ */
+@Internal
+public class TableSchemaUtils {
+
+	public static TableSchema filterGeneratedColumn(TableSchema tableSchema) {
+		Preconditions.checkNotNull(tableSchema);
```

Comment:
We may need to add some document for this method, do you think the name `filterOutGeneratedColumns` is better ?

Reply:
I'm actually +1 to add an interface in `TableSchema` to handle this, but because TableSchema is a public API, so i'm indecisive about that.

Reply:
I think `getPhysicalSchema` is a nice choice